Makes datum stock parts work for Stock parts Experiment purposes

Sealed101 opened the pull request:
## About The Pull Request
Fixes #71845
Sets the machinery scanning experiments up to check for datumised stock parts. 

## Why It's Good For The Game
Scanning module experiment is broken on live due to datumised scanning modules, as the experiment was not modified to check for datum stock parts.

## Changelog
:cl:
fix: fixed Advanced Scanning Modules Calibration Experiment not registering upgraded machines to its progress
/:cl:

Sealed101 committed:
datoom

Mothblocks commented on code/modules/experisci/experiment/types/scanning_machinery.dm:
```diff
@@ -18,9 +18,14 @@
 		return FALSE
 
 	var/obj/machinery/machine = target
+	//this checks for stock parts as objects inside the machine, will be obsolete when stock parts become datumised; please remove this then
```
FWIW Not all stock parts can be datumized out, power cells specifically

Sealed101 replied:
i'll adjust the comment so it doesn't get misinterpreted in the datumised future

Sealed101 committed:
comment tweak
as not all parts will be datumised, we don't need the part
about the obj check that said "will become obsolete"
